### PR TITLE
chore: Remove circular dependency caused by undefinedDropdownValue

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Removes a circular dependency caused by `undefinedDropdownValue` – this PR moves it to a `/constants` folder.

## How to test

A ✅ in the CI should be enough to prove this is good.

Additional circular dependencies exist that mean we can't yet include a linting rule. This will be added in a follow up PR, #267.

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -9,7 +9,8 @@ module.exports = {
       rules: {
         "@typescript-eslint/unbound-method": 0,
         "@typescript-eslint/explicit-module-boundary-types": 0,
-        "@typescript-eslint/no-unused-vars": 2
+        "@typescript-eslint/no-unused-vars": 2,
+        "import/no-cycle": 2
```
Comment (akash1810) [inline .eslintrc.cjs:13]: Any reason to not use [`@guardian/eslint-config-typescript`](https://github.com/guardian/csnx/tree/main/libs/%40guardian/eslint-config-typescript)? Appreciate it'll likely be a bigger change than desired for now.

I'd recommend adding `ignoreExternal` to the `import/no-cycle` config to improve execution time. See https://github.com/guardian/csnx/pull/60.

Comment (jonathonherbert) [reply .eslintrc.cjs]: IIUC we extend that config in the line above – wonder if we're using a version that predates us adding the rule? Be nice to update it in a separate PR.

Comment (akash1810) [reply .eslintrc.cjs]: > IIUC we extend that config in the lines above

Ah! I'll admit to being lazy, and not checking prior to the suggestion!

Comment (jonathonherbert) [reply .eslintrc.cjs]: Ace. We can't yet add that rule in this PR, so I'll resolve this now in anticipation of #267.

**Review by dskamiotis — APPROVED** [review]
This looking good. CI building successfully, however a couple of conflicts to address before merging

**Pull request merged**